plugins/network-elements/nicira-nvp/src/com/cloud/network/resource/NiciraNvpResource.java

```diff
@@ -191,6 +204,19 @@ public PingCommand getCurrentStatus(final long id) {
         return new PingCommand(Host.Type.L2Networking, id);
     }
 
+    private int searchApiProvider(ClusterRoleConfig[] configuredRoles) {
+        int length = configuredRoles.length;
+        for (int i = 0; i < length; i++) {
+            if (! configuredRoles[i].getRole().equals("api_provider")){
```

Comment:
The searchApiProvider returns an index (if the element is found), and that index is then used to obtain the element from the array (Line 190). Why not return the element right away?
This would simplify the code around line 190